web/src/stream_popover.js

```diff
@@ -552,6 +553,20 @@ export async function build_move_topic_to_stream_popover(
         $("#move_topic_form .move_messages_edit_topic").trigger("focus");
     }
 
+    function get_count_of_messages_to_be_moved(selected_option) {
+        if (selected_option === "change_one") {
+            return 1;
+        }
+        if (selected_option === "change_later") {
+            return message_util.get_count_of_messages_in_topic_sent_after_current_message(
+                current_stream_id,
+                topic_name,
+                message.id,
+            );
+        }
+        return message_util.get_messages_in_topic(current_stream_id, topic_name).length;
+    }
```

Comment:
This logic is only correct if `messages_lists.current.data.fetch_status.has_found_newest/oldest` are both true, depending on the mode; i.e. we have all the messages in the the current topic locally cached. This will often be the case, but is not guaranteed to be.

So I think we need to either (1) ask the server if we don't know, which would likely be annoying/expensive with current APIs or (2) have a variant of the wording to say "At least {n} messages will be moved." if the client cannot show it has the complete set.

Reply:
Hey @timabbott, based on my understanding, I think the following approach should be okay:

If `messages_lists.current.data.fetch_status.has_found_newest/oldest` are both true, we can display "N messages will be moved." If not, we can show "At least N messages will be moved."

However, I'm concerned this might be ambiguous for users, as sometimes they will see "N messages" and other times "At least N messages."

Reply:
I guess we would also need to handle the case when the message view is not narrowed to a topic while moving message because in that case `has_found_newest/oldest` fields will not represent the data fetched for that particular topic. Not sure how to do it, maybe we can just show "Atleast N messages ..." for that cases as well.

Reply:
Hey @sahil839, could you suggest if we could limit ourselves to using only "At least N messages..."? I think this would be less ambiguous for users, as they would consistently see "At least N messages..." rather than sometimes seeing "N messages..." and other times "At least N messages...".

Reply:
I am not sure but I feel would be better to just show "N messages ... " when we are sure. @timabbott might provide more feedback.